src/components/map/Map.js

```diff
@@ -42,30 +59,53 @@ class Map extends React.Component {
     map.options.minZoom = 12;
     map.options.maxZoom = 21;
 
+    // Declare the onMarkerfunction so it is locally known and useable in the
+    // then of the getAllBlackspots call
+    const onMarkerClick = this.onMarkerClick;
+
     // Get geo data
     getAllBlackspots()
       .then(geoData => {
         // Add the geo data to the map as markers
         L.geoJSON(geoData, {
           // Add custom markers
           pointToLayer: function(feature, latlng) {
-            console.log(feature.properties.status);
+            // Create a marker with the correct icon and onClick method
+            const { status, spot_type } = feature.properties;
             return L.marker(latlng, {
-              icon: MarkerTypes[feature.properties.spot_type],
-            });
+              icon: L.divIcon({
+                // Add the correct classname based on type
+                // Risici types have a bigger icon therefore need more margin
+                className: `divIconClass ${
+                  spot_type === 'risico' ? 'large' : ''
+                }`,
+                html: renderToString(
+                  <Marker type={spot_type} status={status} />
+                ),
+              }),
+            }).on('click', () => onMarkerClick(feature, latlng, map));
           },
         }).addTo(map);
-
         this.setState({ loading: false });
       })
       .catch(() => {
         this.setState({ error: true, loading: false });
       });
   }
 
+  onMarkerClick(feature, latlng, map) {
+    map.flyTo([latlng.lat, latlng.lng], 14);
```

Comment:
That is fine for now. I shouldn't have merged the detail-panel branch in here since it wasn't done yet. I will fix this in the detail-panel story and branch.